packages/drivers/odsp-driver/src/odspDocumentServiceFactory.ts

```diff
@@ -35,3 +41,62 @@ export class OdspDocumentServiceFactory
         );
     }
 }
+
+/**
+ * Factory for creating sharepoint document service with a provided snapshot.
+ * Use if you don't want to connect to any kind of external/internal storages and want to provide
+ * content directly.
+ */
+export class LocalOdspDocumentServiceFactory extends OdspDocumentServiceFactoryCore {
```

Comment:
maybe move it to its own file? Tree shaking works on a file boundary and is pretty dumb (keeps all exported things), so you will be increasing bundle size for Bohemia with functionality that they do not need.
Similar comment probably applies to everything else - try to separate code that is not needed in interactive apps into its own set of files.
Same logic applies in reverse, though we care much less about bundle size in service.